ForageBuddy/NameReader.cs

```diff
@@ -41,20 +42,33 @@ private static Bitmap TransformImage(Image image, int width, int height)
 
             using (var graphics = Graphics.FromImage(destImage))
             {
-                graphics.CompositingMode = CompositingMode.SourceCopy;
-                graphics.CompositingQuality = CompositingQuality.GammaCorrected;
-                graphics.InterpolationMode = InterpolationMode.Default;
-                graphics.SmoothingMode = SmoothingMode.None;
-                graphics.PixelOffsetMode = PixelOffsetMode.None;
-
                 using (var wrapMode = new ImageAttributes())
                 {
-                    wrapMode.SetWrapMode(WrapMode.Tile);
+                    wrapMode.SetWrapMode(WrapMode.Clamp);
                     graphics.DrawImage(image, destRect, 0, 0, image.Width, image.Height, GraphicsUnit.Pixel, wrapMode);
                 }
             }
 
             return destImage;
         }
+
+        private static Bitmap ToBinaryImage(this Bitmap image)
+        {
+            Bitmap grayScale = new Bitmap(image.Width, image.Height);
+
+            for (int y = 0; y < grayScale.Height; y++)
+                for (int x = 0; x < grayScale.Width; x++)
+                {
+                    Color c = image.GetPixel(x, y);
+
+                    int gs = (int)(c.R * 0.3 + c.G * 0.59 + c.B * 0.11);
+
+                    if (gs > BrigtnessThreshold)
+                        grayScale.SetPixel(x, y, Color.White);
+                    else
+                        grayScale.SetPixel(x, y, Color.Black);
```

Comment:
Can this be:

```
var gs = (int)(c.R * 0.3 + c.G * 0.59 + c.B * 0.11);
var colour = gs > BrigtnessThreshold ? Color.White : Color.Black;

grayScale.SetPixel(x, y, colour);

Reply:
I have an idea for this, give me a moment.

Reply:
Changed, so the image passed in, is over-written.